modules/launcher/src/main/java/org/mule/module/launcher/util/ObservableList.java

```diff
@@ -353,6 +356,7 @@ public void add(E o)
         {
             ObservableList.this.add(o);
             cursor++;
+            lastRet = -1;
```

Comment:
to follow the spec at http://docs.oracle.com/javase/7/docs/api/java/util/ListIterator.html#set(E)

```
void set(E e)

Replaces the last element returned by next() or previous() with the specified element 
(optional operation). This call can be made only if neither remove() nor add(E) have 
been called after the last call to next or previous.
```